crates/rome_js_analyze/src/utils/batch.rs

```diff
@@ -209,52 +215,75 @@ impl JsBatchMutation for BatchMutation<JsLanguage> {
             .unwrap_or(false)
     }
 
-    fn add_jsx_element_after_element(
+    fn add_jsx_elements_after_element<I>(
         &mut self,
         after_element: &AnyJsxChild,
-        new_element: &AnyJsxChild,
-    ) -> bool {
+        new_elements: I,
+    ) -> bool
+    where
+        I: IntoIterator<Item = AnyJsxChild>,
+    {
         let old_list = after_element.parent::<JsxChildList>();
         if let Some(old_list) = &old_list {
             let jsx_child_list = {
                 let mut new_items = vec![];
-                for element in old_list {
-                    new_items.push(element.clone());
-                    if element == *after_element {
-                        new_items.push(new_element.clone());
+                let mut old_elements = old_list.into_iter();
+
+                for old_element in old_elements.by_ref() {
+                    let is_needle = old_element == *after_element;
+
+                    new_items.push(old_element.clone());
+                    if is_needle {
+                        break;
                     }
                 }
 
+                new_items.extend(new_elements);
+                new_items.extend(old_elements);
+
                 jsx_child_list(new_items)
             };
 
-            self.replace_node(old_list.clone(), jsx_child_list);
+            self.replace_node_discard_trivia(old_list.clone(), jsx_child_list);
             true
         } else {
             false
         }
     }
 
-    fn add_jsx_element_before_element(
+    fn add_jsx_elements_before_element<I>(
         &mut self,
-        after_element: &AnyJsxChild,
-        new_element: &AnyJsxChild,
-    ) -> bool {
-        let old_list = after_element.syntax().parent().and_then(JsxChildList::cast);
+        before_element: &AnyJsxChild,
+        new_elements: I,
+    ) -> bool
+    where
+        I: IntoIterator<Item = AnyJsxChild>,
+    {
+        let old_list = before_element
+            .syntax()
+            .parent()
+            .and_then(JsxChildList::cast);
         if let Some(old_list) = &old_list {
             let jsx_child_list = {
                 let mut new_items = vec![];
-                for element in old_list {
-                    if element == *after_element {
-                        new_items.push(new_element.clone());
+                let mut old_elements = old_list.into_iter().peekable();
+
+                while let Some(next_element) = old_elements.peek() {
+                    if next_element == before_element {
+                        break;
                     }
-                    new_items.push(element.clone());
+                    // SAFETY: the peek() makes sure the element exists
+                    let next_element = old_elements.next().unwrap();
+                    new_items.push(next_element);
```

Comment:
Can't we push the peeked element instead so that we avoid the `unwrap` call 

```suggestion
                    new_items.push(next_element);
                    old_elements.next();
```